legend-engine-test-runner-service/src/main/java/org/finos/legend/engine/testable/service/extension/ServiceTestRunner.java

```diff
@@ -164,28 +137,115 @@ else if (service.execution instanceof PureSingleExecution)
         }
     }
 
+    private List<org.finos.legend.engine.protocol.pure.v1.model.test.result.TestResult> executeMultiExecutionTestSuite(PureMultiExecution execution, ServiceTestSuite suite, List<String> testIds, PureModel pureModel, PureModelContextData data, RichIterable<? extends Root_meta_pure_extension_Extension> routerExtensions, MutableList<PlanTransformer> planTransformers, Map<String, MultiExecutionServiceTestResult> testResultsByTestId)
+    {
+        List<Closeable> closeables = null;
+        PureMultiExecution multiExecution = shallowCopyMultiExecution(execution, data);
+        String execkey;
+        try
+        {
+            if (multiExecution.executionParameters != null && !multiExecution.executionParameters.isEmpty())
+            {
+                execkey = multiExecution.executionKey;
+                closeables = multiExecution.executionParameters.stream()
+                        .map(param ->
+                        {
+                            Pair<Runtime, List<Closeable>> closeable = TestRuntimeBuilder.getTestRuntimeAndClosableResources(param.runtime, suite.testData, data);
+                            param.runtime = closeable.getOne();
+                            return closeable.getTwo();
+                        })
+                        .flatMap(Collection::stream)
+                        .collect(Collectors.toList());
+
+            }
+            else
+            {
+                execkey = ((Root_meta_legend_service_metamodel_PureMultiExecution) pureService._execution())._executionKey();
+                List<Closeable> tempCloseables = Lists.mutable.empty();
+                multiExecution.func.body.stream().forEach(valSpec -> valSpec.accept(new TestValueSpecificationBuilder(tempCloseables, suite.testData, data)));
+                closeables = tempCloseables;
+            }
+            ExecutionPlan executionPlan = ServicePlanGenerator.generateCompositeExecutionPlan(multiExecution, null, pureModel, pureVersion, PlanPlatform.JAVA, null, routerExtensions, planTransformers);
+            CompositeExecutionPlan compositeExecutionPlan = (CompositeExecutionPlan) executionPlan;
+            String finalExeckey = execkey;
+            for (Test test : suite.tests)
+            {
+                if (testIds.contains(test.id))
+                {
+                    List<String> keys;
+                    if (((ServiceTest) test).parameters != null)
+                    {
+                        String keyName = ((ServiceTest) test).parameters.stream()
+                                .filter(parameterValue -> parameterValue.name.equals(finalExeckey))
+                                .map(param -> param.value.accept(new PrimitiveValueSpecificationToObjectVisitor())).findFirst().get().toString();
```

Comment:
This line assume the optional always has a value, while integration testing failed with empty value.